rattestation/modsig.c

```diff
@@ -167,8 +167,19 @@ modsig_parse_new(const char *pkcs7_raw, size_t len)
 	X509_ALGOR_get0(&digest_algo_obj, NULL, NULL, digest_algo);
 	X509_ALGOR_get0(&sig_algo_obj, NULL, NULL, sig_algo);
 
-	sig_info->hash_algo = asn1_object_to_hash_algo(digest_algo_obj);
-	sig_info->sig_algo = asn1_object_to_sig_algo(sig_algo_obj);
+	char *hash_algo_name = (char *)malloc(100);
+	char *sig_algo_name = (char *)malloc(100);
+
+	OBJ_obj2txt(hash_algo_name, 100, digest_algo_obj, 0);
+	OBJ_obj2txt(sig_algo_name, 100, sig_algo_obj, 0);
+
+	sig_info->hash_algo = mem_strdup(hash_algo_name);
+	sig_info->sig_algo = mem_strdup(sig_algo_name);
+	mem_free(hash_algo_name);
+	mem_free(sig_algo_name);
```

Comment:
convention: use safe alloc wrapper mem_free0 if possible to detect use after free

Reply:
done